test/functional/p2p_cfilters.py

```diff
@@ -0,0 +1,245 @@
+#!/usr/bin/env python3
+# Copyright (c) 2019 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Tests NODE_COMPACT_FILTERS (BIP 157/158).
+
+Tests that a node configured with -blockfilterindex signals NODE_COMPACT_FILTERS and responds
+correctly to GET_CFILTERS, GET_CFHEADERS, GET_CFCHECKPT requests.
+"""
+
+from test_framework.messages import (
+    FILTER_TYPE_BASIC, NODE_COMPACT_FILTERS,
+    msg_getcfilters, msg_getcfheaders, msg_getcfcheckpt,
+    ser_uint256, uint256_from_str, hash256,
+)
+from test_framework.mininode import (
+    P2PInterface,
+)
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    connect_nodes, disconnect_nodes, sync_blocks,
+    wait_until,
+)
+
+FILTER_TYPES = ["basic"]
+
+class CFiltersClient(P2PInterface):
+    def __init__(self):
+        super().__init__()
+        # Store the cfilters received.
+        self.cfilters = []
+
+    def pop_cfilters(self):
+        cfilters = self.cfilters
+        self.cfilters = []
+        return cfilters
+
+    def on_cfilter(self, message):
+        """Store cfilters received in a list."""
+        self.cfilters.append(message)
+
+class CompactFiltersTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
```

Comment:
in commit a7b375a4469fcef823a274e3f05db0f9ab7aa035 : 

The test still times out on valgrind. Please bump the timeout.

```diff
diff --git a/test/functional/p2p_cfilters.py b/test/functional/p2p_cfilters.py
index dca35f4e89..d16946510b 100755
--- a/test/functional/p2p_cfilters.py
+++ b/test/functional/p2p_cfilters.py
@@ -42,6 +42,7 @@ class CFiltersClient(P2PInterface):
 
 class CompactFiltersTest(BitcoinTestFramework):
     def set_test_params(self):
+        self.rpc_timeout = 480
         self.setup_clean_chain = True
         self.num_nodes = 2
         self.extra_args = [
```

(generating more than 1000 blocks in valgrind takes time)


https://travis-ci.org/github/bitcoin/bitcoin/jobs/682347754#L3147


```
test_framework.authproxy.JSONRPCException: 'generatetoaddress' RPC took longer than 120.000000 seconds. Consider using larger timeout for calls that take longer to return. (-344)